File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.py
class Solution:
  def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
    for i in range(1, len(arr)):
      arr[i] ^= arr[i - 1]
    ans = []
    for l,r in queries:
      ans.append((arr[l - 1] if l else 0) ^ arr[r])
    return ans

File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.php
class Solution {
  function xorQueries($arr, $queries) {
    for ($i = 1; $i < count($arr); $i++) {
      $arr[$i] ^= $arr[$i - 1];
    }
    $ans = [];
    foreach ($queries as list($l, $r)) {
      $ans[] = ($l ? $arr[$l - 1] : 0) ^ $arr[$r];
    }
    return $ans;
  }
}